File: Go/Count-Negative-Numbers-In-A-Sorted-Matrix.go
package main

func countNegatives(grid [][]int) int {
	var count int

	for _, inner := range grid {
		for i := len(inner) - 1; i >= 0; i-- {
			if inner[i] >= 0 {
				break
			}
			count++
		}
	}
	return count
}

File: Rust/Count-Negative-Numbers-In-A-Sorted-Matrix.rs
impl Solution {
    pub fn count_negatives(grid: Vec<Vec<i32>>) -> i32 {
        let mut count = 0;

        for outer in grid {
            for num in outer {
                if num.is_negative() {
                    count += 1
                }
            }
        }
        count
    }
}
